governance/proposals/010-arbitrum-l1-l2-messaging.js

```diff
@@ -0,0 +1,225 @@
+    ARBTokenAddressOnL2,
+    ERC20_ABI,
+    l2Wallet
+  ).connect(l2Wallet)
+
+  const balanceOf = await L2TokenContract.balanceOf(timelockL2Alias)
+  const tokenAmount = ethers.parseEther('1')
+
+  // Create an instance of the Interface from the ABIs
+  const iface_erc20 = new ethers.Interface(ERC20_ABI)
+  const iface_inbox = new ethers.Interface(INBOX_ABI)
+
+  // Encode the ERC20 Token transfer calldata
+  const transfer_calldata = iface_erc20.encodeFunctionData('transfer', [
+    grantsContractAddress,
+    tokenAmount,
+  ])
+  /**
+   * Now we can query the required gas params using the estimateAll method in Arbitrum SDK
+   */
+  const l1ToL2MessageGasEstimate = new L1ToL2MessageGasEstimator(l2Provider)
+
+  /**
+   * The estimateAll method gives us the following values for sending an L1->L2 message
+   * (1) maxSubmissionCost: The maximum cost to be paid for submitting the transaction
+   * (2) gasLimit: The L2 gas limit
+   * (3) deposit: The total amount to deposit on L1 to cover L2 gas and L2 call value
+   */
+  const L1ToL2MessageGasParams = await l1ToL2MessageGasEstimate.estimateAll(
+    {
+      from: L1TimelockContract,
+      to: ARBTokenAddressOnL2,
+      l2CallValue: 0,
+      excessFeeRefundAddress: l2Wallet.address,
+      callValueRefundAddress: l2Wallet.address,
+      data: transfer_calldata,
+    },
+    await getBaseFee(l1Provider),
+    l1Provider
+  )
+  const gasPriceBid = await l2Provider.getGasPrice()
+
+  const inbox_calldata = iface_inbox.encodeFunctionData(
+    'createRetryableTicket',
+    [
+      ARBTokenAddressOnL2,
+      0,
+      L1ToL2MessageGasParams.maxSubmissionCost,
+      l2Wallet.address,
+      l2Wallet.address,
+      L1ToL2MessageGasParams.gasLimit,
+      gasPriceBid,
+      transfer_calldata,
+    ]
+  )
+
+  const proposalName = `# Test Transaction before 7k ARB Transfer To Fund Unlock Protocol’s Ecosystem via Grants Stack  This proposal requests to use 1 ARB from the tokens given to Unlock Protocol DAO by ArbitrumDAO to run a test transaction to de-risk the transfer of 7k ARB tokens to fund the retroQF round on Grants Stack.`
+  // Proposal ARGS i.e Call Governor.propose() directly with these values
+  const targets = [inboxAddress]
+  const values = [L1ToL2MessageGasParams.deposit.toNumber() * 10] // I Multiply by 10 to add extra in case gas changes
+  const calldatas = [inbox_calldata]
+  const description = proposalName
+
+  const calls = [
+    {
+      contractNameOrAbi: INBOX_ABI,
+      contractAddress: inboxAddress,
+      functionName: 'createRetryableTicket',
+      functionArgs: [
+        ARBTokenAddressOnL2,
+        0,
+        L1ToL2MessageGasParams.maxSubmissionCost,
+        l2Wallet.address,
+        l2Wallet.address,
+        L1ToL2MessageGasParams.gasLimit,
+        gasPriceBid,
+        transfer_calldata,
+      ],
+      value: L1ToL2MessageGasParams.deposit.toNumber() * 10, // I Multiply by 10 to add extra in case gas changes due to proposal delay
```

Comment:
Are you sure you need to pass the value again here? It is already declared above

Reply:
Yes, from my understanding of the existing format, and correct me if I'm wrong but this value property is responsible for passing the eth amount to a payable function, and in this case that's the eth deposit to be sent to arbitrum's inbox contract. 
If you were referring to this line ```const values = [L1ToL2MessageGasParams.deposit.toNumber() * 10]``` that was declared to show how the proposal can be created from an interface like etherscan or remix, but if its not necessary or confusing I can remove it as well.